rmf_task/src/rmf_task/requests/Clean.cpp

```diff
@@ -23,92 +23,77 @@ namespace rmf_task {
 namespace requests {
 
 //==============================================================================
-class CleanDescription::Implementation
+class Clean::Model : public Request::Model
 {
 public:
 
-  Implementation()
-  {}
-
-  std::size_t start_waypoint;
-  std::size_t end_waypoint;
-  rmf_traffic::Trajectory cleaning_path;
-  rmf_battery::ConstMotionPowerSinkPtr motion_sink;
-  rmf_battery::ConstDevicePowerSinkPtr ambient_sink;
-  rmf_battery::ConstDevicePowerSinkPtr cleaning_sink;
-  std::shared_ptr<const rmf_traffic::agv::Planner> planner;
-  rmf_traffic::Time start_time;
-  bool drain_battery;
-
-  rmf_traffic::Duration invariant_duration;
-  double invariant_battery_drain;
+  std::optional<Estimate> estimate_finish(
+    const agv::State& initial_state,
+    const agv::Constraints& task_planning_constraints,
+    EstimateCache& estimate_cache) const final;
+
+  rmf_traffic::Duration invariant_duration() const final;
+
+  Model(
+    rmf_traffic::Time earliest_start_time,
+    agv::Parameters parameters,
+    rmf_traffic::Trajectory cleaning_path,
+    std::size_t start_waypoint,
+    std::size_t end_waypoint);
+
+private:
+  rmf_traffic::Time _earliest_start_time;
+  agv::Parameters _parameters;
+  rmf_traffic::Trajectory& _cleaning_path;
```

Comment:
We shouldn't use a raw mutable reference here. This is begging for trouble.

If the goal is to avoid duplicate `Trajectory` allocations, we should either have the `Model` take in a `std::shared_ptr<Clean::Description>` that gives it access to the trajectory, or we should have `Clean::Description` use a `std::shared_ptr<Trajectory>` that it can pass along to the `Model`.

Reply:
Realized `_cleaning_path` is unused so got rid of it. https://github.com/open-rmf/rmf_task/pull/15/commits/6490a3635f3d1c1e7e9339223de61eb29e18481b